src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -59,20 +61,24 @@ internal class TaskActivityShim : TaskActivity
             FunctionResult result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
             if (!result.Succeeded)
             {
+                // Flow the original activity function exception to the orchestration
+                // without the outer FunctionInvocationException.
+                Exception exceptionToReport = StripFunctionInvocationException(result.Exception);
+
                 this.config.TraceHelper.FunctionFailed(
                     this.config.HubName,
                     this.activityName,
                     this.activityVersion,
                     instanceId,
-                    result.Exception?.ToString() ?? string.Empty,
+                    exceptionToReport?.ToString() ?? string.Empty,
                     functionType: FunctionType.Activity,
                     isReplay: false);
 
-                if (result.Exception != null)
+                if (exceptionToReport != null)
```

Comment:
>null [](start = 41, length = 4)

If for some reason the strip failed, should we log internally what the exception was?

Reply:
I don't think we actually ever expect `exceptionToReport` to be null. I'm not sure what you mean when you say "for some reason the strip failed". Can you elaborate?

---
In reply to: [181521966](https://github.com/Azure/azure-functions-durable-extension/pull/250#discussion_r181521966) [](ancestors = 181521966)

Reply:
I was referring to the method StripFunctionInvocationException. If we expect 

e as FunctionInvocationException to never return null, then it's OK.

---
In reply to: [181523045](https://github.com/Azure/azure-functions-durable-extension/pull/250#discussion_r181523045) [](ancestors = 181523045,181521966)

Reply:
Right - I'm saying I don't think that method will ever return null. `e as FunctionInvocationException` might return null, but in that case we return `e` directly, so there shouldn't be any nulls. I can remove the `?` operator if that would make things less confusing.

---
In reply to: [181523423](https://github.com/Azure/azure-functions-durable-extension/pull/250#discussion_r181523423) [](ancestors = 181523423,181523045,181521966)

Reply:
Actually, I'll probably keep the null checks as they are. I don't have any guarantee that `result.Exception` will be non-null when `result.Succeeded` is false since that depends on the behavior of WebJobs. But to your original concern, I don't think there is ever a case where this is an exception and we fail to log it.

---
In reply to: [181523733](https://github.com/Azure/azure-functions-durable-extension/pull/250#discussion_r181523733) [](ancestors = 181523733,181523423,181523045,181521966)